src/api/define.js

```diff
@@ -102,10 +102,19 @@ function generateUniqueName(name) {
 }
 
 function registerUniqueName(name) {
-  registry[name] = registry[name] ? registry[name] + 1 : 1;
+  if (registry[name]) {
+    registerUniqueName(name + '-'+ registry[name]);
```

Comment:
If you're using HMR in dev mode and you haven't refreshed the page, but it's reloaded x number of times, this has x - 1 levels of recursion. What was the problem with the previous implementation?